Userland/Libraries/LibWeb/WebAudio/AnalyserNode.cpp

```diff
@@ -0,0 +1,329 @@
+    auto frequency_domain_dat = apply_a_fourier_transform(blackman_windowed_input);
+
+    // FIXME: 4. Smooth over time the frequency domain data.
+    auto smoothed_data = smoothing_over_time(frequency_domain_dat);
+
+    // FIXME: 5. Convert to dB.
+    return conversion_to_dB(smoothed_data);
+}
+
+// https://webaudio.github.io/web-audio-api/#dom-analysernode-getfloatfrequencydata
+WebIDL::ExceptionOr<void> AnalyserNode::get_float_frequency_data(JS::Handle<WebIDL::BufferSource>& array)
+{
+
+    dbgln("FIXME: Analyser node: implement getFloatFrequencyData");
+    // from https://webaudio.github.io/web-audio-api/#conversion-to-db
+    // This array, 𝑌[𝑘], is copied to the output array for getFloatFrequencyData().
+    // array = current_frequency_data();
+    (void)array;
+    return {};
+}
+
+// https://webaudio.github.io/web-audio-api/#dom-analysernode-getbytefrequencydata
+WebIDL::ExceptionOr<void> AnalyserNode::get_byte_frequency_data(JS::Handle<WebIDL::BufferSource>& array)
+{
+    // FIXME: If another call to getByteFrequencyData() or getFloatFrequencyData() occurs within the same render
+    // quantum as a previous call, the current frequency data is not updated with the same data. Instead,
+    // the previously computed data is returned.
+    //      Need to implement some kind of blocking mechanism, I guess
+    //      Might be more obvious how to handle this when render quantums have some
+    //      more scaffolding
+    //
+
+    // Write the current frequency data into array. If array’s byte length is less than frequencyBinCount,
+    // the excess elements will be dropped. If array’s byte length is greater than the frequencyBinCount ,
+    // the excess elements will be ignored. The most recent fftSize frames are used in computing the frequency data.
+
+    // current_frequency_data returns a vector of size m_fftSize
+    Vector<double> dB_data = current_frequency_data();
+
+    // For getByteFrequencyData(), the 𝑌[𝑘] is clipped to lie between minDecibels and maxDecibels
+    // and then scaled to fit in an unsigned byte such that minDecibels is represented by the
+    // value 0 and maxDecibels is represented by the value 255.
+
+    // FIXME: Naive
+    for (auto x : dB_data) {
+        if (x < m_min_decibels)
+            x = m_min_decibels;
+
+        if (x > m_max_decibels)
+            x = m_max_decibels;
+    }
+
+    Vector<u8> byte_data;
+    byte_data.ensure_capacity(dB_data.size());
+
+    double delta_dB = m_max_decibels - m_min_decibels;
+    for (size_t i = 0; i < dB_data.size(); i++)
+        byte_data[i] = u8(255 * (dB_data[i] - m_min_decibels) / delta_dB);
+
+    // FIXME: need to copy data into Uint8Array
+    //        need to manage the spec vs containers you can do math on
+    //        also need to account for the fftSize frames requirements above
+    // array = byte_data;
+    (void)array;
+    return {};
```

Comment:
I think you can implement this once you take `array` as a `JS::Uint8Array` like I suggested.